src/test/scala/tech/cryptonomic/nautilus/cloud/adapters/doobie/DoobieMeteringStatsRepositoryTest.scala

```diff
@@ -0,0 +1,78 @@
+
+import java.util.concurrent.TimeUnit
+
+import org.scalatest.{BeforeAndAfterEach, EitherValues, Matchers, OneInstancePerTest, OptionValues, WordSpec}
+import tech.cryptonomic.nautilus.cloud.adapters.metering.MeteringApiConfig
+import tech.cryptonomic.nautilus.cloud.domain.metering.stats.AggregatedMeteringStats
+import tech.cryptonomic.nautilus.cloud.fixtures.Fixtures
+import tech.cryptonomic.nautilus.cloud.tools.{DefaultNautilusContext, InMemoryDatabase}
+
+import scala.concurrent.duration._
+import scala.concurrent.duration.FiniteDuration
+
+class DoobieMeteringStatsRepositoryTest extends WordSpec
+  with Matchers
+  with EitherValues
+  with OptionValues
+  with InMemoryDatabase
+  with BeforeAndAfterEach
+  with Fixtures
+  with OneInstancePerTest {
+
+  val context = DefaultNautilusContext
+  val sut = context.meteringStatsRepository
+  val config = MeteringApiConfig(
+    "http",
+    "loclalhost",
+    1234,
+    5.seconds,
+    FiniteDuration(5, TimeUnit.MINUTES),
+    FiniteDuration(5, TimeUnit.MINUTES),
+    "key"
+  )
+
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+
+    applySchemaWithFixtures()
+    context.userRepository.createUser(exampleCreateUser).unsafeRunSync()
+    context.userRepository.createUser(exampleCreateUser.copy(userEmail = "another@domain.com")).unsafeRunSync()
+  }
+
+  "DoobieMeteringStatsRepository" should {
+    "insert and read aggregated stats" in {
+      // given
+      val exampleAggregateMeteringStats = List(AggregatedMeteringStats(exampleUser.userId, "dev", 1, time5m.minusSeconds(config.statsInterval.toSeconds), time5m))
+      sut.insertStats(exampleAggregateMeteringStats).unsafeRunSync()
+      // when
+      val result = sut.getStatsPerUser(exampleUser.userId)
+      // then
+      result.unsafeRunSync() should contain theSameElementsAs exampleAggregateMeteringStats
+    }
+    "not insert duplicated stats" in {
+      // given
+      val exampleAggregateMeteringStats = List(AggregatedMeteringStats(exampleUser.userId, "dev", 1, time5m.minusSeconds(config.statsInterval.toSeconds), time5m))
+      sut.insertStats(exampleAggregateMeteringStats).flatMap(_ => sut.insertStats(exampleAggregateMeteringStats)).unsafeRunSync()
+      // when
+      val result = sut.getStatsPerUser(exampleUser.userId)
+      // then
+      result.unsafeRunSync() should contain theSameElementsAs exampleAggregateMeteringStats
+    }
+    "get correct statistics by timestamp" in {
+      // given
+      val exampleAggregateMeteringStats = List(
+        AggregatedMeteringStats(exampleUser.userId, "dev", 1, time5m.minusSeconds(config.statsInterval.toSeconds - 300), time5m.minusSeconds(300)),
+        AggregatedMeteringStats(exampleUser.userId, "dev", 1, time5m.minusSeconds(config.statsInterval.toSeconds + 300), time5m.plusSeconds(300)),
+        AggregatedMeteringStats(exampleUser.userId, "dev", 1, time5m.minusSeconds(config.statsInterval.toSeconds ), time5m)
```

Comment:
we get list of the intervals from the MeteringAPI, for given API key they will be touch(not really an overlap) on begin/end of two consecutive intervals.

Reply:
I see 3 intervals, and the last one is the easier to grasp, it's `[time5m - statsInterval, time5m]` where `time5m` is an arbitrary reference point in time.

So it's an interval as wide as the stats reference window, that ends at `time5m`.
It's not immediately apparent but it's quite straightforward to figure out.

The other two intervals require mental gymn to understand what is 300, where to add it and where to remove it... I suppose one is 300 seconds before the last and the other 300 after? But I need to really squint to understand it and make ballpark math.
Is 300 a magic number? Is it related to the statsInterval?

All these questions arise because the definition is not straightforward to read, it's the computational result of what the test creator had in mind.
To make it easy to understand, I can suggest a few things like
- give the intervals a name as a `val`
- use names for other constants like `300seconds` that possibly clarify why the choice of 300 related to other existing measures
- try to define a local function to "shift" or "resize" intervals, such that a new interval might be easier to relate to the others

e.g.
```scala
val middleInterval = interval(end = time5m, wide=statsInterval)
val firstInterval = shiftBefore(middleInterval, 300.seconds)
val lastInterval = after(middleInterval, wide=statsInterval)
```

Reply:
These are no-rule by any means, it's just my personal stance on test readability, where my goal would be to allow anyone needing to verify the test logic and possibly add new tests similar to this one, to spend no more than a couple minutes to read the test, without doing any mental computation to deduct the test assumptions.
This for anyone new to the application, or anyone of us after a few months after having not touched this code.
YMMV

Reply:
TBH maybe I've encoded in my mind that 300 is 5minutes when I see this magic number is my instant association.
I've change it to more be more descriptive, I did not feel it required to introduce new functions, at least for now.